File: templates/tutorial/python/fastapi/src/api/routes/health.py
"""Health check routes for FastAPI application.

FastAPI route handlers are defined with decorators like @app.get().
The decorator specifies the HTTP method (GET, POST, etc.) and URL path.
"""
from fastapi import APIRouter, HTTPException

from src.core.config import settings


# APIRouter groups related routes - think of it like a chapter for API endpoints
router = APIRouter(tags=["health"])


@router.get("/")
def health() -> dict[str, str]:
    """Health check endpoint - returns status and current environment.

    Returns:
        dict: A dictionary that FastAPI automatically converts to JSON
    """
    return {"status": "ok", "env": settings.env}


@router.get("/readyz")
def ready() -> dict[str, str]:
    """Readiness check - used by load balancers to know if the app is ready."""
    return {"status": "ready", "project": "{{project_name}}"}

File: initra/templates/go/gin/internal/handlers/health.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"{{project_name}}/internal/config"
)

// Health responds with a liveness payload.
func Health(cfg config.Config) gin.HandlerFunc {
	return func(c *gin.Context) {
		c.JSON(http.StatusOK, gin.H{
			"status":  "ok",
			"project": "{{project_name}}",
			"env":     cfg.Env,
		})
	}
}
